### ISSUE TITLE
StackOverflowError in transformPatternsArithmeticExpressions when commutative operator has 2 inline arguments

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14187]
In the event of a commutative operator having 2 inline arguments (e.g. `1 + 2`), the `Settings.transformPatternsArithmeticExpressions` pattern transformation will produce a `StackOverflowError`:

```
java.lang.StackOverflowError
	at org.jooq.impl.QOM$UCommutativeOperator.$swap(QOM.java:6942)
	at org.jooq.impl.Patterns.lambda$4(Patterns.java:1049)
	at org.jooq.impl.QOM.replaceUntilStable(QOM.java:7980)
	at org.jooq.impl.QOM.replace(QOM.java:7205)
	at org.jooq.impl.QOM$UOperator2.$replace(QOM.java:6824)
	at org.jooq.impl.QOM.replaceUntilStable(QOM.java:7982)
	at org.jooq.impl.QOM.replace(QOM.java:7205)
	at org.jooq.impl.QOM$UOperator2.$replace(QOM.java:6824)
	at org.jooq.impl.QOM.replaceUntilStable(QOM.java:7982)
	at org.jooq.impl.QOM.replace(QOM.java:7205)
```

The transformation tries to normalise the expression:

```sql
x + 1 => 1 + x
```

But of course, this should only be done if only 1 argument is an inline value. `Settings.transformPrecalculations` (see https://github.com/jOOQ/jOOQ/issues/13593) would prevent this.

This was discovered while testing `transformPatternsNullOnNullInput` (where we have `null + 1`, which are also 2 inline arguments)

**Comment by lukaseder** [comment]
Related: https://github.com/jOOQ/jOOQ/issues/12903

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0 and 3.17.6 (https://github.com/jOOQ/jOOQ/issues/14188)
